CITATION.cff

```diff
@@ -0,0 +1,13 @@
+abstract: A sample CLI tool made in V that prints geometric shapes to the screen.
+authors:
+  - alias: hungrybluedev
+    family-names: Haldar
+    given-names: Subhomoy
+cff-version: 1.2.0
+date-released: 2023-04-20
```

Comment:
### Updating `date-released`

This field follows the regular expression `^date-released: ....-..-..` and can thus be updated very easily with:

```
sed "s/^date-released: ....-..-../date-released: $(date +%F)/g" CITATION.cff > tmp && mv tmp CITATION.cff
```

I also made a GitHub Action around this command:  https://github.com/kevinmatthes/cff-release-today.

If you should happen to already use Aeruginous for something in the release preparation workflow (e.g. assembling your CHANGELOG or increment the version numbers), you can also call `aeruginous cff-release-today CITATION.cff`.  This approach as well targets the regular expression and will use the current day of the runner (usually UTC's "today").

Using the regular expression and UTC-today is the most effective option because it has low overhead regarding the implementation and the timezone referred to for international communication is usually UTC.

### Updating `version`

Here, you should use some version incrementing system like bump2version (unfortunately [discontinued](https://github.com/c4urself/bump2version/issues/268)) or Aeruginous `increment-version` and explicitly target the top level `version` field in order to prevent editing the references' versions by accident.

Reply:
Nice! I believe `v bump` will be accomodate the version update process too. It has no file restrictions.

Reply:
@hungrybluedev

I just tested to bump the version with `v bump --patch CITATION.cff` and noticed that the result is not correct:  the value of `cff-version` was incremented but not this project's `version`.  Although the CI -- in fact, the CFF linter -- would easily catch this error, it still points out that there might be a bug.

I also tested to bump the versions of the CITATION.cffs of the following projects:

- https://github.com/kevinmatthes/aeruginous-rs (Rust CLI toolbox)
- https://github.com/kevinmatthes/adjust (V TUI text editor (WIP); the project where I cite *this* project)

In case of Aeruginous, again, only the CFF version was incremented.  In contrast to the other two cases, nothing happened for Adjust although `v bump` always returned with `Bumped version in CITATION.cff`.

Since both projects are owned by me, this is no problem for me because I use my Aeruginous to bump the versions (together with some other automation).  But for other users, this behaviour of `v bump` might cause problems, so I thought you should rather know about it as you are the author.  I decided to mention this problem here because we already discussed bumping versions of CITATION.cffs in this PR.

Reply:
Hey!

I'm a bit busy this week but I will definitely come back and look at it in the weekend.

IIRC, v bump needs to have all target files specified.

Reply:
@kevinmatthes I tried the command you have provided. I've noticed that cff-version is incremented, and not version. Was your intention to increment the latter and not the former?

If so, I can implement a finer targeting mechanism for version bumping.

Reply:
@hungrybluedev

You are right, I intended to increment the value of `version` and not of `cff-version` but by accident, the software incremented the `cff-version` field.

Reply:
@kevinmatthes I've made a PR to add a `--skip` flag to `v bump`. You can clone my fork of V and test the feature until it gets merged officially.

Reply:
Link to PR: https://github.com/vlang/v/pull/19064

Reply:
@hungrybluedev

Thank you for your work on this!